src/aof.c

```diff
@@ -792,6 +792,38 @@ int openNewIncrAofForAppend(void) {
     return C_OK;
 }
 
+/* Called in `rewriteAppendOnlyFileBackground` to open a new temp INCR AOF when in 
+ * AOF_WAIT_REWRITE state, the AOF information will not be added to server.aof_manifest 
+ * and persisted to manifest file.
+ * */
+int openNewTempIncrAofForAppend() {
+    serverAssert(server.aof_manifest != NULL);
+    serverAssert(server.aof_state == AOF_WAIT_REWRITE);
+
+    /* Dup a temp aof_manifest to modify. */
+    aofManifest *temp_am = aofManifestDup(server.aof_manifest);
+
+    /* Open new AOF. */
+    sds new_aof_name = getNewIncrAofName(temp_am);
+    sds new_aof_filepath = makePath(server.aof_dirname, new_aof_name);
+    int newfd = open(new_aof_filepath, O_WRONLY|O_TRUNC|O_CREAT, 0644);
```

Comment:
Do we open all AOF related files with read access to all? Isn't this a security issue?

Reply:
looks like this is what Redis always did, so from that i can conclude:
1. we didn't introduce any security regression recently.
2. if it was a problem, maybe someone would have opened an issue long ago.
3. if we'll change it now, it'll probably break many usages and cause some mess (but on the other hand that's always the nature of security hardening)

@yossigo WDYT?